cmd/verify/rosa/cmd.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright (c) 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rosa
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/PuerkitoBio/goquery"
+	"github.com/hashicorp/go-version"
+	"github.com/openshift/rosa/pkg/info"
+	"github.com/openshift/rosa/pkg/reporter"
+	"github.com/spf13/cobra"
+)
+
+var Cmd = &cobra.Command{
+	Use:     "rosa-client",
+	Aliases: []string{"rosa"},
+	Short:   "Verify ROSA client tools",
+	Long:    "Verify that the ROSA client tools is installed and compatible.",
+	Example: `  # Verify rosa client tools
+  rosa verify rosa`,
+	Run: run,
+}
+
+const (
+	DownloadLatestMirrorFolder = "https://mirror.openshift.com/pub/openshift-v4/clients/rosa/latest/"
+	BaseReleasesFolder         = "https://mirror.openshift.com/pub/openshift-v4/clients/rosa"
+)
+
+func run(_ *cobra.Command, _ []string) {
+	rprtr := reporter.CreateReporterOrExit()
+	resp, err := http.Get(BaseReleasesFolder)
+	if err != nil {
+		rprtr.Errorf("Error setting up request for latest released rosa cli: %v", err)
+		os.Exit(1)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		rprtr.Errorf("Error while requesting latest released rosa cli: %d %s", resp.StatusCode, resp.Status)
+		os.Exit(1)
+	}
+	doc, err := goquery.NewDocumentFromReader(resp.Body)
+	if err != nil {
+		rprtr.Errorf("Error parsing response body: %v", err)
+		os.Exit(1)
+	}
+	possibleVersions := []string{}
```

Comment:
NP
We can decompose conditional lines 64-75 to a new function **findPossibleVersions**?